homeassistant/components/hvac/zwave.py

```diff
@@ -0,0 +1,244 @@
+
+    @property
+    def should_poll(self):
+        """No polling on ZWave."""
+        return False
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan speed set."""
+        return self._current_operation_state
+
+    @property
+    def fan_list(self):
+        """List of available fan modes."""
+        return self._fan_list
+
+    @property
+    def current_swing_mode(self):
+        """Return the fan speed set."""
+        return self._current_swing_mode
+
+    @property
+    def swing_list(self):
+        """List of available swing modes."""
+        return self._swing_list
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        unit = self._unit
+        if unit == 'C':
+            return TEMP_CELCIUS
+        elif unit == 'F':
+            return TEMP_FAHRENHEIT
+        else:
+            return unit
+        return self.hass.config.temperature_unit
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def current_operation(self):
+        """Return the current operation mode."""
+        return self._current_operation
+
+    @property
+    def operation_list(self):
+        """List of available operation modes."""
+        return self._operation_list
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    def set_temperature(self, temperature):
+        """Set new target temperature."""
+        # pylint: disable=too-many-branches
+        if self._zxt_120 == 1:
+            # ZXT-120 does not support get setpoint
+            self._target_temperature = temperature
+            if self._current_operation == 'Heat':
+                for value in self._node.get_values(class_id=0x43).values():
+                    if value.command_class == 67 and value.index == 1:
+                        # ZXT-120 only responds to whole int
+                        value.data = int(round(temperature, 0))
+            elif self._current_operation == 'Cool':
+                for value in self._node.get_values(class_id=0x43).values():
+                    if value.command_class == 67 and value.index == 2:
+                        value.data = int(round(temperature, 0))
+            elif self._current_operation == 'Dry Air':
+                for value in self._node.get_values(class_id=0x43).values():
+                    if value.command_class == 67 and value.index == 8:
+                        value.data = int(round(temperature, 0))
+            elif self._current_operation == 'Auto Changeover':
+                for value in self._node.get_values(class_id=0x43).values():
+                    if value.command_class == 67 and value.index == 10:
+                        value.data = int(round(temperature, 0))
```

Comment:
Is the ```value_changed``` callback called automatically even when changing the value from the home assistant side, ie not from zwave side? Otherwise, the state in home assistant won't be updated, and you would have to call self.update_ha_state(True) here.